elmslie-core/src/main/java/vivid/money/elmslie/core/store/Actor.kt

```diff
@@ -0,0 +1,32 @@
+package vivid.money.elmslie.core.store
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.flow.onEach
+import vivid.money.elmslie.core.config.ElmslieConfig
+
+fun interface Actor<Command : Any, Event : Any> {
```

Comment:
extensions are not accessible within SAM implementations

we need either make it an abstract class with non-overridable flow, throwable and event extensions (making two of them private) or let it be fun interface, but extract extensions out of it, once again, making 2 of them private and aiming to use flow extension with the context receiver one day (🙏🏻 for kotlin 2.1)  

the same goes for the RxActor

Reply:
yeah, but in practice, does it matter? I can't image (only for sample purpose maybe) production Actor as a SAM impl, usually you want to provide some deps to your Actor, so you'll have to use inheritance. 
Can we leave just interface then?

Reply:
With interface inheritance, these extensions can be applied only inside class, we can leave them public)

Reply:
it's a dirty api then imo, leaving them inside interface, thus allowing an override, which is not intended

Reply:
Ok, got it.  I guess 'abstract class' way will be more proper, since I want these extensions be used only inside Actor as intended. Moreover, I suppose making abstract class gives the opportunity for RxActor inherits Actor and wrap its extensions with RxJava. What's think?)

Reply:
I failed with inheritance)

Reply:
I made commit with abstract class, check please

Reply:
nice